src/Sulu/Bundle/MediaBundle/Controller/CollectionController.php

```diff
@@ -368,14 +369,16 @@ protected function saveEntity($id, Request $request)
             $collectionManager = $this->getCollectionManager();
             $data = $this->getData($request);
             $data['id'] = $id;
-            $data['locale'] = $this->getLocale($request);
+            $data['locale'] = $this->getRequestParameter($request, 'locale', true);
```

Comment:
That's because at this point the locale is really required and the process should fail if its not given. In other methods the locale is not necessarily required and can also be null, which is the default behavior of `getLocale` in the RestController.